models/baseModels/InvoiceItem/InvoiceItem.ts

```diff
@@ -191,15 +191,29 @@ export abstract class InvoiceItem extends Doc {
     },
     transferUnit: {
       formula: async (fieldname) => {
+        if (!this.item) return '';
         if (fieldname === 'quantity' || fieldname === 'unit') {
           return this.unit;
         }
 
-        return (await this.fyo.getValue(
-          'Item',
-          this.item as string,
-          'unit'
-        )) as string;
+        const conversionItems = await this.fyo.db.getAll(
+          ModelNameEnum.UOMConversionItem,
+          {
+            fields: ['uom'],
+            filters: { parent: this.item },
+          }
+        );
+
+        if (conversionItems.length === 0) {
+          return this.unit;
+        }
```

Comment:
~if (conversionItems.length === 0)~

`if(conversionItems.length)` 

might do well. You don't have to specifically check the length in numbers if you just want to check the existence.